paras-marketplace-contract/src/lib.rs

```diff
@@ -433,7 +433,7 @@ impl Contract {
             "Paras: NEAR support only"
         );
 
-        assert!(market_data.is_auction.is_none(), "Paras: the NFT is on auction");
+        assert_eq!(market_data.is_auction.unwrap(), false, "Paras: the NFT is on auction");
```

Comment:
please check : 

if is_auction.is_none() -> allow buy
if is_auction.is_some() && is_auction.unwrap() == false -> allow buy

use if let Some() .. for cleaner code